tests/syntax_error/self_take_arg.jou

```diff
@@ -0,0 +1,8 @@
+class Foo:
+    def bar(self, x: int) -> void:
+        return
+
+def main() -> int:
+    f = Foo{}
+    f.bar(1, 2) # Error: method bar(self: Foo*, x: int) takes 1 argument, but it was called with 2 arguments
```

Comment:
I think it would be nicer if the error message didn't include `self: Foo*`. Instead of `bar(self: Foo*, x: int)` it would say `bar(x: int)`, because when calling a method, you only need to care about the `x`, not about `self`.

To do this, you can pass `False` in self-hosted compiler and add a similar parameter to `signature_to_string()` in the other compiler.

Reply:
okay